erigon-lib/state/inverted_index.go

```diff
@@ -832,33 +835,42 @@ func (iit *InvertedIndexRoTx) Prune(ctx context.Context, rwTx kv.RwTx, txFrom, t
 		}
 
 		stat.PruneCountTx++
-		// This DeleteCurrent needs to the last in the loop iteration, because it invalidates k and v
-		if err = rwTx.Delete(ii.indexKeysTable, k); err != nil {
+		if err := idxAncientsCollector.Collect(k, nil); err != nil {
 			return nil, err
 		}
 
 		if ctx.Err() != nil {
 			return nil, ctx.Err()
 		}
 	}
+	// Close cursor to avoid spilling garbage to disk.
+	keysCursor.Close()
+	if err := idxAncientsCollector.Load(nil, "", func(key, txnm []byte, table etl.CurrentTableReader, next etl.LoadNextFunc) error {
+		if err := rwTx.Delete(iit.ii.indexKeysTable, key); err != nil {
```

Comment:
hmmm, next question if it really helps not spill pages, since txnums already ordered